core/math/src/main/java/org/arakhne/afc/math/geometry/d2/afp/BasicPathShadow2afp.java

```diff
@@ -184,19 +188,15 @@ private void discretizePathIterator(
             return;
         }
         PathElement2afp element;
-
         element = pi.next();
         if (element.getType() != PathElementType.MOVE_TO) {
             throw new IllegalArgumentException(Locale.getString(Path2afp.class, "E1")); //$NON-NLS-1$
         }
-
-        Path2afp<?, ?, ?, ?, ?, ?> localPath;
         double movx = element.getToX();
         double movy = element.getToY();
         double curx = movx;
         double cury = movy;
-        double endx;
-        double endy;
+        this.coordinateParam = new BasicPathShadow2afp.CoordinatesParam(x1, y1, x2, y2, curx, cury);
```

Comment:
The CoordinateParam should be creating inside the loop. Indeed curx and cury are changing over the loops.